### ISSUE TITLE
[Bug]: Streaming not working

### ISSUE THREAD

**Issue by goldenfaeron** [run-llama/llama_index#12118]
### Bug Description

For me event the most minimal starter does not stream anymore. I worked with llamaindex before and everything was working as expected. But since today I don't get a stream output with the response_gen

I hope I have not done any silly mistake

### Version

0.10.20

### Steps to Reproduce

1. install latest llama_index version

2. Run the example code from the docs:
```
from dotenv import load_dotenv
load_dotenv()

from llama_index.core import VectorStoreIndex, SimpleDirectoryReader

documents = SimpleDirectoryReader("data").load_data()
index = VectorStoreIndex.from_documents(documents)
query_engine = index.as_query_engine(streaming=True)
response_stream = query_engine.query(
    "What did the author do growing up?",
)
for text in response_stream.response_gen:
    print(text)
```

### Relevant Logs/Tracbacks

```shell
I don't get any output
```

**Comment by goldenfaeron** [comment]
@dosu-bot the query_engine is a BaseQueryEngine not a RetrieverQueryEngine with streaming enabled. therefore response_gen exists

**Comment by goldenfaeron** [comment]
@dosu-bot 
everything is working well but this:
for text in streaming_response.response_gen:
    print(text)
    
I dont get any prints using this.
When i do `print(streaming_response.response_txt)` it works well and prints the finished answer

**Comment by goldenfaeron** [comment]
@dosu-bot
i use the default OpeAI LLM with the key in a .env
this is the whole code i am running:

```
from llama_index.core import VectorStoreIndex, SimpleDirectoryReader
from dotenv import load_dotenv
load_dotenv()

documents = SimpleDirectoryReader("data").load_data()
index = VectorStoreIndex.from_documents(documents)
query_engine = index.as_query_engine(streaming=True)

streaming_response = query_engine.query("What did the author do growing up?")
for text in streaming_response.response_gen:
    print(text)
```

I get no prints back

**Comment by juleskuehn** [comment]
Yes, this seems broken in 10.20.post2 ; downgrading to 10.20.post1 solved it for me

**Comment by goldenfaeron** [comment]
seems fixed with llama-index-core==10.21.post1

**Comment by willhamlam** [comment]
> Yes, this seems broken in 10.20.post2 ; downgrading to 10.20.post1 solved it for me

Thanks bro, I've debugging this streaming problem for an hour... until I saw your comment. 🙏
